### DISCUSSION TITLE
Display time in a different timezone?

### DISCUSSION THREAD

**Discussion by cx-lukas-salkauskas-x** [wezterm/wezterm#1095] [Q&A] [answered]
Hi there,
I want to display my current time and other time in different timezone.

I know that a shell command `TZ="US/Pacific" date` gives the output I need.

```lua
wezterm.on("update-right-status", function(window, pane)
  local date = wezterm.strftime("%a %b %-d %H:%M:%S %Y ");
  local bat = ""
  local _, sftime, _ = wezterm.run_child_process({"date"});
  for _, b in ipairs(wezterm.battery_info()) do
    bat = "🔋 " .. string.format("%.0f%%", b.state_of_charge * 100)
  end

  window:set_right_status(wezterm.format({
    {Text=bat .. "   Local: "..date.."| US/Pacific: "..sftime},
  }));
end)
```

So either how to do that with `run_child_process` or maybe there is an other way?

**Comment by kar1** [comment, marked as answer]
Lua only reports the time of the operating system. The time docs are here: http://www.lua.org/manual/5.2/manual.html#pdf-os.date

You could take that time as UTC and add the offset for US/Pacific -7:
```lua
os.date("!%a %b %d %H:%M:%S %Y ", os.time() - 7 * 60 * 60)
```

But that doesn't adjust for different summer time offsets if any. For a more complete Lua solution you could install a Lua module like: https://github.com/daurnimator/luatz

For the `date` command you could try this:
```lua
wezterm.run_child_process({"/bin/sh", "-c", "TZ=US/Pacific date"});
```

**Reply by wez** [reply]
FWIW, another option for running date is: `wezterm.run_child_process({"env", "TZ=US/Pacific", "date"})`

**Comment by wez** [comment]
I think it's a reasonable feature request for `wezterm.strftime` to accept an optional timezone parameter.  The underlying rust chrono crate supports timezones (and names using https://docs.rs/chrono-tz/0.5.3/chrono_tz/), and the end result would be significantly cheaper than shelling out to `date` in the context of updating the right status bar.
